library/libwaku.nim

```diff
@@ -39,6 +42,17 @@ const RET_MISSING_CALLBACK: cint = 2
 ################################################################################
 ### Not-exported components
 
+
+template foreignThreadGc(body: untyped) =
```

Comment:
I use this template when executing the callbacks instead of having a single call. We need to ensure that Nim's GC can properly manage memory allocated by Nim code that might be running in other language threads.